packages/backend/src/ee/services/ai/tools/generateBarVizConfig.ts

```diff
@@ -1,48 +1,48 @@
-import { DynamicStructuredTool } from '@langchain/core/tools';
-import {
-    AiWebAppPrompt,
-    getErrorMessage,
-    isSlackPrompt,
-    SlackPrompt,
-    UpdateSlackResponse,
-} from '@lightdash/common';
+import { getErrorMessage, isSlackPrompt } from '@lightdash/common';
 import * as Sentry from '@sentry/node';
-import { z } from 'zod';
-import { SlackClient } from '../../../../clients/Slack/SlackClient';
+import { tool } from 'ai';
 import Logger from '../../../../logging/logger';
-import { renderEcharts } from '../charts/renderEcharts';
+import { renderEcharts } from '../../AiService/charts/renderEcharts';
 import {
     generateBarVizConfigToolSchema,
     renderVerticalBarMetricChart,
-} from '../charts/verticalBarChart';
-import { RunMiniMetricQuery } from '../runMiniMetricQuery/runMiniMetricQuery';
+} from '../../AiService/charts/verticalBarChart';
+import type {
+    GetPromptFn,
+    RunMiniMetricQueryFn,
+    SendFileFn,
+    UpdateProgressFn,
+    UpdatePromptFn,
+} from '../types/aiAgentDependencies';
 
-type GetGenerateBarVizConfigToolArgs = {
-    updateProgress: (progress: string) => Promise<void>;
-    runMiniMetricQuery: RunMiniMetricQuery;
-    getPrompt: () => Promise<SlackPrompt | AiWebAppPrompt>;
-    updatePrompt: (prompt: UpdateSlackResponse) => Promise<void>;
-    sendFile: InstanceType<typeof SlackClient>['postFileToThread'];
+type Dependencies = {
+    updateProgress: UpdateProgressFn;
+    runMiniMetricQuery: RunMiniMetricQueryFn;
+    getPrompt: GetPromptFn;
+    updatePrompt: UpdatePromptFn;
+    sendFile: SendFileFn;
 };
```

Comment:
(could be outside of the scope of this pr) Let's consider passing some kind of context variable to each tool, could come in handy later when we might want to pass some state around or tracing

```
type ToolDependencies<T> = { ctx: ToolContext } & T
```